### PR TITLE
implement get selection range

### PR DESCRIPTION
**Pull request by DanielVu08**
using in SWARM-1076

### CODE REVIEW

**Comment by DanielVu08** [comment]
@KakitLee Hi JJ, can you help me merge this. Thanks

File: RichEditorView/Assets/editor/rich_editor.js [hunk 1]
```diff
@@ -480,3 +480,27 @@ RE.getRelativeCaretYPosition = function() {
     
     return y;
 };
+
+RE.getSelectionCharOffsetsWithin = function() {
+    var start = 0, end = 0;
```
Comment (KakitLee) [inline RichEditorView/Assets/editor/rich_editor.js]: let start = 0, end = 0;

File: RichEditorView/Assets/editor/rich_editor.js [hunk 2]
```diff
@@ -480,3 +480,27 @@ RE.getRelativeCaretYPosition = function() {
...
+    var sel, range, priorRange;
```
Comment (KakitLee) [inline RichEditorView/Assets/editor/rich_editor.js]: same as above

File: RichEditorView/Assets/editor/rich_editor.js [hunk 3]
```diff
@@ -480,3 +480,27 @@ RE.getRelativeCaretYPosition = function() {
...
+    } else if (typeof document.selection != "undefined" &&
+               (sel = document.selection).type != "Control") {
+        range = sel.createRange();
+        priorRange = document.body.createTextRange();
+        priorRange.moveToElementText(element);
+        priorRange.setEndPoint("EndToStart", range);
+        start = priorRange.text.length;
+        end = start + range.text.length;
+    }
+    var string = start + ":" + end;
```
Comment (KakitLee) [inline RichEditorView/Assets/editor/rich_editor.js]: return start + ":" + end;

**Review by KakitLee — COMMENTED** [review]
Reviewed. Just change the code and use your same branch to send another PR.

File: RichEditorView/Assets/editor/rich_editor.js (hunk 3 above)
Comment (DanielVu08) [reply RichEditorView/Assets/editor/rich_editor.js]: let it in a 'string' var, later it will be easier if format needs changing to suit model

Comment (KakitLee) [reply RichEditorView/Assets/editor/rich_editor.js]: OK. But change var to let, please use ES6 syntax

**Review by KakitLee — APPROVED** [review]

**Comment by DanielVu08** [comment]
@KakitLee thanks bro :D

**Comment by DanielVu08** [comment]
@KakitLee please merge it when you're done,  I don't have merge permission.

**Pull request merged**